src/components/Auth/api.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RestApi } from '../common/rest_api';
+import { AddAuthUserPayload, AuthUser } from './types';
+
+export interface ApiAuthUserFields {
+  passwordHash: string;
+}
+
+export default class AuthApi extends RestApi {
+  private readonly baseUrl = `http://${this.hostAndPort}/identitytoolkit.googleapis.com/v1/`;
+  private readonly baseUrlWithProject =
+    this.baseUrl + `projects/${this.projectId}/accounts:`;
+  private readonly baseUrlWithoutProject = this.baseUrl + `accounts:`;
```

Comment:
There is no need for this `withoutProject` version. All APIs used below are also available under the project prefix (preferred for admin requests).